src/validationinterface.h

```diff
@@ -34,6 +35,8 @@ class CValidationInterface {
     virtual void Inventory(const uint256 &hash) {}
     virtual void ResendWalletTransactions(int64_t nBestBlockTime) {}
     virtual void BlockChecked(const CBlock&, const CValidationState&) {}
+    virtual void GetScriptForMining(CScript &script) {};
```

Comment:
I tend to disagree that it's the responsibility of the listeners to handle what to do when there are multiple listeners - you may not always have control over in which order they are invoked, and that can result in hard to debug problems.

I'm not sure we even want to have multiple subscribers here, ever? If not, a combiner could simply throw an error if multiple return values are returned.